File: solutions/26.ts
let maxRem = 0;
let maxDiv = 0;

for (let divisor = 2; divisor <= 1000; divisor++) {
  let dividend = 1;
  const remainders: number[] = [];

  while (dividend % divisor !== 0) {
    while (dividend / divisor < 1) {
      dividend *= 10;
    }

    const partialQ = Math.floor(dividend / divisor);
    const remainder = dividend % divisor;

    if (remainders.includes(remainder)) break;

    remainders.push(remainder);
    dividend -= divisor * partialQ;
  }

  if (remainders.length > maxRem) {
    maxRem = remainders.length;
    maxDiv = divisor;
  }
}

console.log(maxDiv);
// 983

File: solutions/26.swift
import Foundation

var maxRem = 0
var maxDiv = 0
for divisor in 2..<1000 {
  var dividend = 1
  var quotient: [Int] = []
  var remainders: [Int] = []

  while dividend % divisor != 0 {
    while dividend / divisor < 1 {
      dividend *= 10
    }

    let partialQ = dividend / divisor
    let remainder = dividend % divisor

    if remainders.contains(remainder) {
      break
    }

    remainders.append(remainder)
    quotient.append(partialQ)
    dividend -= divisor * partialQ
  }

  if remainders.count > maxRem {
    maxRem = remainders.count
    maxDiv = divisor
  }
}

print(maxDiv)
// 983
